Issue #31: amp: bucket replication

danielcahoon opened the pull request.

Commit:
Issue #31: amp: bucket replication

icahoon commented on pkg/amp/bucket_sets.go:
```diff
@@ -0,0 +1,111 @@
+
+type bucketSets struct {
+	*resource.Resources
+	BucketSets []*config.BucketSet
```
Is this required? It looks like the config for the bucket sets is only used in the constructor.

icahoon commented on pkg/amp/bucket_sets.go:
```diff
@@ -0,0 +1,111 @@
+
+	b := &bucketSets{
+		Resources:  resource.NewResources(),
+		BucketSets: cfg,
```
This can probably be removed

icahoon commented on pkg/aws/bucket.go:
```diff
@@ -124,56 +124,32 @@ func (b *bucket) enableEncryption() error {
+			Rules: []*s3.ReplicationRule{
+				{
+					Destination: &s3.Destination{
+						Bucket: aws.String("arn:aws:s3:::" + b.Destination()),
```
should you be using the arn code here rather than hardcoding it?

icahoon commented on pkg/aws/bucket.go:
```diff
@@ -124,56 +124,32 @@ func (b *bucket) enableEncryption() error {
+					Destination: &s3.Destination{
+						Bucket: aws.String("arn:aws:s3:::" + b.Destination()),
 					},
+					Prefix: aws.String(""),
```
What does Prefix default to? Perhaps this isn't needed?

icahoon reviewed (changes requested):
This is looking great. Just some minor comments.

Commit:
Issue #31: amp: bucket replication

- fixed the buckets to use *config.Buckets instead of []*config.Bucket
- fixed the bucket sets to use *config.BucketSets instead of
[]*config.BucketSet
- made the bucket arn more clear in pkg/aws/bucket.go for
enableReplication()

Commit:
Issue #31: amp: bucket replication

- fixed the copyright

icahoon reviewed (approved):
lgtm. thanks for making the changes.

icahoon commented:
this closes issue #31
